soc/nordic/nrf53/soc.c

```diff
@@ -54,22 +54,14 @@
 #define HFXO_NODE DT_NODELABEL(hfxo)
 
 /* LFXO config from DT */
+#if DT_NODE_HAS_STATUS_OKAY(LFXO_NODE)
+BUILD_ASSERT(DT_NODE_HAS_PROP(LFXO_NODE, load_capacitors));
 #if DT_ENUM_HAS_VALUE(LFXO_NODE, load_capacitors, external)
 #define LFXO_CAP NRF_OSCILLATORS_LFXO_CAP_EXTERNAL
 #elif DT_ENUM_HAS_VALUE(LFXO_NODE, load_capacitors, internal)
 #define LFXO_CAP (DT_ENUM_IDX(LFXO_NODE, load_capacitance_picofarad) + 1U)
-#else
-/* LFXO config from legacy Kconfig */
-#if defined(CONFIG_SOC_LFXO_CAP_INT_6PF)
-#define LFXO_CAP NRF_OSCILLATORS_LFXO_CAP_6PF
-#elif defined(CONFIG_SOC_LFXO_CAP_INT_7PF)
-#define LFXO_CAP NRF_OSCILLATORS_LFXO_CAP_7PF
-#elif defined(CONFIG_SOC_LFXO_CAP_INT_9PF)
-#define LFXO_CAP NRF_OSCILLATORS_LFXO_CAP_9PF
-#else
-#define LFXO_CAP NRF_OSCILLATORS_LFXO_CAP_EXTERNAL
-#endif
-#endif
```

Comment:
This part cannot be removed yet. These options are deprecated but they are still present in Kconfig, so they still need to be supported.

Reply:
Yes, they were deprecated in 4.0, so the can be removed now.